src/batched/KokkosBatched_ApplyQ_Serial_Impl.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef __KOKKOSBATCHED_APPLY_Q_SERIAL_IMPL_HPP__
+#define __KOKKOSBATCHED_APPLY_Q_SERIAL_IMPL_HPP__
+
+
+/// \author Kyungjoo Kim (kyukim@sandia.gov)
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_ApplyQ_Serial_Internal.hpp"
+
+namespace KokkosBatched {
+
+  ///
+  /// Serial Impl
+  /// ===========
+
+  template<>
```

Comment:
Why is there the empty `template<>` followed by another `template<...>`?

Reply:
It is partial specialization. The first template argument is struct template allowing partial specialization and the followed template arguments are function templates.